packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -248,40 +248,45 @@ export async function collectMetadata(
 export async function accumulateMetadata(
   metadataItems: MetadataItems
 ): Promise<ResolvedMetadata> {
-  const deepClone: <T>(v: T) => T =
-    process.env.NODE_ENV === 'development'
-      ? require('next/dist/compiled/@edge-runtime/primitives/structured-clone')
-          .structuredClone
-      : (v) => v
   const resolvedMetadata = createDefaultMetadata()
 
-  let parentPromise = Promise.resolve(resolvedMetadata)
-
+  const promises: Promise<ResolvedMetadata>[] = [
+    Promise.resolve(resolvedMetadata),
+  ]
   for (const item of metadataItems) {
```

Comment:
one semantic thing that go lost in recent refactors but is probably a mistake.

Page metadata is not at a lower layer than Layout metadata. This is important when considering which title template applies.

```
/app/layout.js -> template "%s - root"
/app/deep/layout.js -> template "%s - deep"
/app/deep/page.js -> title "page"
```
we want to see the rendered titlef or /deep/page as `"page - root"` not `"page - deep"`.

I believe the way things currently stand we would end up with `page - deep`. This might require having both Layout and Page metadata on teh same metadataItem instance. The nice thing is that it will always be the last item b/c there are no items deeper than Page so we can do our special logic once our iterator hits the last pass through

Reply:
maybe accumulateMetadata sig is `(metadataItems: MetadataItems, pageMetadataItem?: MetadataItem)`

Reply:
actually that doesn't work b/c we need to pair the page metadata with the layout but we don't know if the last item in metadataItems is the same layer or not

Reply:
maybe metadtatItem is
```
{

  metadataExport,
  pageMetadataExport?,
  staticFilesMetadata,
}
```